### PR TITLE
api: refactor to use early-return pattern

### PR DESCRIPTION
**Pull request by eduardoveiga**
This commit apply early-return pattern in namespace

### CODE REVIEW

File: api/nsadm/service.go [hunk 1]
```diff
@@ -60,13 +61,22 @@ func (s *service) ListNamespaces(ctx context.Context, pagination paginator.Query
...
+		default:
+			return nil, err
+		}
 	}
+
 	namespace.Name = strings.ToLower(namespace.Name)
-	ns, _ := s.store.NamespaceGetByName(ctx, namespace.Name)
-	if ns != nil {
+	otherNamespace, err := s.store.NamespaceGetByName(ctx, namespace.Name)
+	if err != nil && err != store.ErrNoDocuments {
```
Comment (leonardojoao) [inline api/nsadm/service.go:76]: What do you think add "switch case" this case?

File: api/nsadm/service.go [hunk 2]
```diff
@@ -91,98 +101,202 @@ func (s *service) GetNamespace(ctx context.Context, namespace string) (*models.N
...
-		if OwnerUser, _ := s.store.UserGetByID(ctx, ownerID); OwnerUser != nil && OwnerUser.Username != username {
-			if ns.Owner == OwnerUser.ID {
-				if user, _ := s.store.UserGetByUsername(ctx, username); user != nil {
-					if ns, err := s.store.NamespaceRemoveMember(ctx, namespace, user.ID); err == nil {
-						return ns, err
-					}
-				}
-				return nil, ErrUserNotFound
-			}
+	_, err := s.store.NamespaceGet(ctx, namespace)
```
Comment (leonardojoao) [inline api/nsadm/service.go:237]: if _, err := s.store.NamespaceGet(ctx, namespace); err != nil { ... }

File: api/nsadm/service.go [hunk 3]
```diff
@@ -91,98 +101,202 @@ func (s *service) GetNamespace(ctx context.Context, namespace string) (*models.N
...
+			return nil, ErrNamespaceNotFound
+
+		default:
+			return nil, err
 		}
-		return nil, ErrUnauthorized
 	}
-	return nil, ErrNamespaceNotFound
+
+	_, err = s.store.UserGetByID(ctx, ownerID)
```
Comment (leonardojoao) [inline api/nsadm/service.go:248]: if _, err = s.store.UserGetByID(ctx, ownerID); err != nil { ... }

File: api/nsadm/service.go [hunk 4]
```diff
@@ -91,98 +101,202 @@ func (s *service) GetNamespace(ctx context.Context, namespace string) (*models.N
...
+		case store.ErrNoDocuments:
+			return nil, ErrUserNotFound
+
+		default:
+			return nil, err
+		}
+	}
+
+	ns, err := s.store.NamespaceRemoveMember(ctx, namespace, user.ID)
+	return ns, err
```
Comment (leonardojoao) [inline api/nsadm/service.go:271]: return s.store.NamespaceRemoveMember(ctx, namespace, user.ID)

File: api/nsadm/service.go [hunk 5]
```diff
@@ -91,98 +101,202 @@ func (s *service) GetNamespace(ctx context.Context, namespace string) (*models.N
...
 }
 
 func (s *service) UpdateDataUserSecurity(ctx context.Context, sessionRecord bool, tenant string) error {
-	ns, _ := s.GetNamespace(ctx, tenant)
-	if ns != nil {
-		return s.store.NamespaceSetSessionRecord(ctx, sessionRecord, tenant)
+	_, err := s.GetNamespace(ctx, tenant)
```
Comment (leonardojoao) [inline api/nsadm/service.go:275]: if _, err := s.GetNamespace(ctx, tenant); err != nil { ... }

File: api/nsadm/service.go [hunk 6]
```diff
@@ -91,98 +101,202 @@ func (s *service) GetNamespace(ctx context.Context, namespace string) (*models.N
...
-	return ErrUnauthorized
+
+	return s.store.NamespaceSetSessionRecord(ctx, sessionRecord, tenant)
 }
 
 func (s *service) GetDataUserSecurity(ctx context.Context, tenant string) (bool, error) {
-	ns, _ := s.GetNamespace(ctx, tenant)
-	if ns != nil {
-		return s.store.NamespaceGetSessionRecord(ctx, tenant)
+	_, err := s.store.NamespaceGet(ctx, tenant)
```
Comment (leonardojoao) [inline api/nsadm/service.go:290]: if _, err := s.store.NamespaceGet(ctx, tenant); err != nil { ... }

File: api/nsadm/service.go [hunk 7]
```diff
@@ -91,98 +101,189 @@ func (s *service) GetNamespace(ctx context.Context, namespace string) (*models.N
...
+
+	return s.store.NamespaceSetSessionRecord(ctx, sessionRecord, tenant)
 }
 
 func (s *service) GetDataUserSecurity(ctx context.Context, tenant string) (bool, error) {
-	ns, _ := s.GetNamespace(ctx, tenant)
-	if ns != nil {
-		return s.store.NamespaceGetSessionRecord(ctx, tenant)
+	if _, err := s.store.NamespaceGet(ctx, tenant); err != nil {
+
```
Comment (leonardojoao) [inline api/nsadm/service.go:279]: You should remove this line.

**Review by otavio — CHANGES_REQUESTED** [review]
@eduardoveiga, the error handling code is very repetitive.

It seems we could refactor it in a similar way we did in the store package, providing a single function that handles the errors, avoiding this handling to be duplicated in almost every service handler.

**Pull request merged**